Framework/Utils/test/test_RawParser.cxx

```diff
@@ -37,10 +41,24 @@ BOOST_AUTO_TEST_CASE(test_RawParser)
     auto* data = reinterpret_cast<size_t*>(buffer.data() + pageNo * PageSize + rdh->headerSize);
     *data = pageNo;
   }
+}
+
+// FIXME: want to run the check for different header versions but when introducing
+// a template parameter, there is a compiler error for using get_if
+//
+// test_RawParser.cxx:72:30: error: expected primary-expression before ‘>’ token
+//     BOOST_CHECK(it.get_if<RDH>() != nullptr);
+template <typename T>
```

Comment:
@ktf, do you have an idea what can be wrong? If I make this a templated function, the `get_if` does not compile anymore.
```
Framework/Utils/test/test_RawParser.cxx:77:30: error: expected primary-expression before ‘>’ token
     BOOST_CHECK(it.get_if<RDH>() != nullptr);
```
It's the same if isolated from `BOOST_CHECK` macro. And also if I specify a concrete version instead of the template argument. So it seems, only specifying a template argument definition breaks already the compilation.

Reply:
@matthiasrichter I believe the right way to get templatized boost checks is to use BOOST_AUTO_TEST_CASE_TEMPLATE. At least this is how I've done it in e.g. https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/MUON/MCH/Raw/Encoder/Payload/testElinkEncoder.cxx

Reply:
thanks for the hint, but it looks like its something different, still scratching my head.